src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,118 @@
+import mate.academy.springboot.swagger.mapper.RequestDtoMapper;
+import mate.academy.springboot.swagger.mapper.ResponseDtoMapper;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.service.ProductService;
+import mate.academy.springboot.swagger.service.util.SortRequestParser;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final RequestDtoMapper<ProductRequestDto, Product> requestMapper;
+    private final ResponseDtoMapper<Product, ProductResponseDto> responseMapper;
+
+    public ProductController(ProductService productService,
+                             RequestDtoMapper<ProductRequestDto, Product> requestMapper,
+                             ResponseDtoMapper<Product, ProductResponseDto> responseMapper) {
+        this.productService = productService;
+        this.requestMapper = requestMapper;
+        this.responseMapper = responseMapper;
+    }
+
+    @PostMapping("/create")
+    @ApiOperation("Endpoint to create new products")
+    public ProductResponseDto create(@RequestBody @Valid ProductRequestDto productRequestDto) {
+        return responseMapper.mapToDto(productService.create(
+                requestMapper.mapToModel(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation("Endpoint to get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return responseMapper.mapToDto(productService.get(id)
+                .orElseThrow(() -> new NoSuchElementException("Can`t find product with id:"
+                        + id)));
+    }
+
+    @GetMapping("/all")
+    @ApiOperation("Endpoint to get all products with pagination and ability "
+            + "to sort by price or by title in ASC or DESC order")
+    public List<ProductResponseDto> findAllSortByPrice(@ApiParam("default value = 20")
+                                                       @RequestParam(defaultValue = "20")
+                                                       Integer count,
+                                                       @ApiParam("default value = 0")
+                                                       @RequestParam(defaultValue = "0")
+                                                       Integer page,
```

Comment:
Let's correct a bit ApiParam annotation usage format:
```suggestion
    public List<ProductResponseDto> findAllSortByPrice(@ApiParam(value = "page size", defaultValue = "20")
                                                       @RequestParam(defaultValue = "20")
                                                       Integer count,
                                                       @ApiParam(value = "page number", defaultValue = "0")
                                                       @RequestParam(defaultValue = "0")
                                                       Integer page,
```
Fix all similar cases please.